docs/user_guides/fs/feature_group/on_demand_transformations.md

```diff
@@ -0,0 +1,180 @@
+    ```python
+
+    feature_vector = feature_view.get_feature_vector(entry={"id":1}, request_parameter={"arg1":1, "arg2":2})
+    ```
+
+#### Retrieving feature vectors
+
+The `get_feature_vectors` function retrieves multiple feature vectors using a list of serving keys. The `request_parameter` in this case, can be a list of dictionaries that specifies the input parameters for the computation of on-demand features for each serving key or can be a dictionary if the on-demand transformations require the same parameters for all serving keys.
+
+=== "Python"    
+!!! example "Recomputing on-demand features while retrieving a feature vectors"
+    ```python
+
+    # Specify unique request parameters for each serving key.
+    feature_vector = feature_view.get_feature_vectors(entry=[{"id":1}, {"id":2}], request_parameter=[{"arg1":1, "arg2":2}, {"arg1":11, "arg2":12}])
+
+    # Specify common request parameters for all serving key.
+    feature_vector = feature_view.get_feature_vectors(entry=[{"id":1}, {"id":2}], request_parameter={"arg1":1, "arg2":2})
+    ```
+
+The `get_feature_vector` and `get_feature_vectors` can also return untransformed features by setting the parameter `transformed` to `False`.
+
+=== "Python"    
+!!! example "Returning untransformed feature vectors"
+    ```python
+
+    untransformed_feature_vector = feature_view.get_feature_vector(entry={"id":1}, transformed=False)
+
+    untransformed_feature_vectors = feature_view.get_feature_vectors(entry=[{"id":1}, {"id":2}], transformed=False)
+    ```
+
+#### Recompute all on-demand features
+
+The `compute_on_demand_features` function recomputes all on-demand features attached to a feature view and adds them to the feature vectors provided as input to the function. This function does not apply model-dependent transformations to any of the features. The `transform` function can be used to apply model-dependent transformations to the returned values if required.
+
+The `request_parameter` in this case, can be a list of dictionaries that specifies the input parameters for the computation of on-demand features for each feature vector given as input to the function or can be a dictionary if the on-demand transformations require the same parameters for all input feature vectors.
+
+=== "Python"    
+!!! example "Recomputing all on-demand features and manually applying model dependent transformations."
+    ```python
+
+    # Specify request parameters for each serving key.
+    untransformed_feature_vector = feature_view.get_feature_vector(entry={"id":1}, transformed=False)
+
+    # re-compute and add on-demand features to the feature vector
+    feature_vector_with_on_demand_features = fv.compute_on_demand_features(untransformed_feature_vector, request_parameter={"arg1":1, "arg2":2})
+
+    # Applying model dependent transformations
+    encoded_feature_vector = fv.transform(feature_vector_with_on_demand_features)
```

Comment:
`fv.transform` the naming is a little bit confusing to me. in `get_feature_vector(entry={"id":1}, transformed=False)` `transformed=True` means it applies **BOTH** on demand transformation and model dependent transformation, but `fv.transform` is only applying MDT. 
I would expect to get a complete transformed feature by calling `fv.transform` using the input of `get_feature_vector(entry={"id":1}, transformed=False)`.

maybe call it `model_dependent_transform` or similar?